src/components/JaegerIntegration/JaegerResults/SpanTable.tsx

```diff
@@ -1,243 +1,250 @@
-    return '/namespaces/' + namespace + '/workloads/' + workload + '?tab=logs';
+  private renderSummary = (span: Span) => {
+    switch (getSpanType(span)) {
+      case 'envoy':
+        return this.renderEnvoySummary(span);
+      case 'http':
+        return this.renderHTTPSummary(span);
+      case 'tcp':
+        return this.renderTCPSummary(span);
+    }
+    // Unknown
+    return this.renderCommonSummary(span);
   };
 
-  private getNodeLog = (sp: Span) => {
-    const srv = sp.process.serviceName.split('.')[0];
-    let workloadNs = getWorkloadFromSpan(sp);
-    if (!workloadNs && this.props.jaegerInfo && this.props.jaegerInfo.whiteListIstioSystem.includes(srv)) {
-      // Special case (why?)
-      workloadNs = {
-        workload: srv === 'jaeger-query' ? 'jaeger' : srv,
-        namespace: serverConfig.istioNamespace
-      };
+  private renderCommonSummary(span: Span) {
+    const info = extractOpenTracingBaseInfo(span);
+    return (
+      <>
+        <div>
+          <strong>Operation: </strong>
+          {span.operationName}
+        </div>
+        <div>
+          <strong>Component: </strong>
+          {info.component || 'unknown'}
+        </div>
+      </>
+    );
+  }
+
+  private renderEnvoySummary(span: Span) {
+    const info = extractEnvoySpanInfo(span);
+    let rqLabel = 'Request';
+    let peerLink: JSX.Element | undefined = undefined;
+    if (info.direction === 'inbound') {
+      rqLabel = 'Received request';
+      if (info.peer && info.peerNamespace) {
+        peerLink = (
+          <>
+            {' from '}
+            <Link to={'/namespaces/' + info.peerNamespace + '/workloads/' + info.peer}>{info.peer}</Link>
+          </>
+        );
+      }
+    } else if (info.direction === 'outbound') {
+      rqLabel = 'Sent request';
+      if (info.peer && info.peerNamespace) {
+        peerLink = (
+          <>
+            {' to '}
+            <Link to={'/namespaces/' + info.peerNamespace + '/services/' + info.peer}>{info.peer}</Link>
+          </>
+        );
+      }
+    }
+    const rsDetails: string[] = [];
+    if (info.statusCode) {
+      rsDetails.push(String(info.statusCode));
     }
-    if (workloadNs) {
-      const path = this.goLogsWorkloads(workloadNs.workload, workloadNs.namespace);
-      return (
-        <Tooltip content={<>View logs of workload {workloadNs.workload}</>}>
-          <Link to={path} onClick={() => history.push(path)}>
-            View logs
-          </Link>
-        </Tooltip>
-      );
+    if (info.responseFlags) {
+      rsDetails.push(info.responseFlags);
     }
 
-    return <>Logs not found</>;
-  };
+    return (
+      <>
+        {this.renderCommonSummary(span)}
+        <div>
+          <strong>
+            {rqLabel}
+            {peerLink}:{' '}
+          </strong>
+          {info.method} {info.url}
+        </div>
+        <div>
+          <strong>Response status: </strong>
+          {rsDetails.join(', ')}
+        </div>
```

Comment:
Should we have errorCodes with red like before? I don't know if it's useful for the user

Reply:
Yes, I wrote some TODO somewhere, I keep that in mind, +1